server/routes/logic.js

```diff
@@ -0,0 +1,123 @@
+require('dotenv').config();
+const bcrypt = require('bcrypt');
+const saltRounds = 10;
+const db = require('../db/db'); // ../db/db is based on how far from index.js
+const jwt = require('jsonwebtoken');
+const cookie = require("cookie");
+
+// Check if the user is already authenticated
+const routeAccessibility = async (req, res, next) => {
+  const authCookie = req.cookies.AccessToken;
+  if (authCookie) {
+    jwt.verify(authCookie, process.env.ACCESS_TOKEN_SECRET, (err, user) => { // user is a value serialised wich is authenticatedUser
+      if (err) return res.sendStatus(403);
+      res.redirect('/authenticated');
+    });
+  } else {  
+    next();
+  }
+};
+
+// Check user existense
+const signIn =  async (req, res) => {
+  if (req.body.user_name) {
+    const name = req.body.user_name;
+    const pw = req.body.password;
+    const data = await db.checkIn(name);
+    if (data) {
```

Comment:
This data can also be an error too I believe: https://github.com/snkzt/user-information-database/pull/7/files#diff-0e0c962c17d2dbb509faf124ae28f7a93341ee047ca1ebace5e1d0ee6f9d831eR24

You will have to fix your error handling as your program can crash here if an error is sent and then you try to get the `data.user_id` from the error object on line 28

Reply:
But, on line 30 you try to retrieve the `data.err`.  Have you taken a look at the data object?  Does it actually contain this field?

I will need some more clarification on this part of the code.